bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/productcarousel/ProductCarouselImpl.java

```diff
@@ -0,0 +1,141 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.adobe.cq.commerce.core.components.internal.models.v1.productcarousel;
+
+
+import com.adobe.cq.commerce.core.components.internal.models.v1.MagentoGraphqlClient;
+import com.adobe.cq.commerce.core.components.internal.models.v1.Utils;
+import com.adobe.cq.commerce.core.components.internal.models.v1.productlist.ProductListItemImpl;
+import com.adobe.cq.commerce.core.components.models.productcarousel.ProductCarousel;
+import com.adobe.cq.commerce.core.components.models.productlist.ProductListItem;
+import java.util.Arrays;
+import com.adobe.cq.commerce.graphql.client.GraphqlClient;
+import com.adobe.cq.commerce.graphql.client.GraphqlResponse;
+import com.adobe.cq.commerce.magento.graphql.FilterTypeInput;
+import com.adobe.cq.commerce.magento.graphql.Operations;
+import com.adobe.cq.commerce.magento.graphql.ProductFilterInput;
+import com.adobe.cq.commerce.magento.graphql.ProductInterface;
+import com.adobe.cq.commerce.magento.graphql.ProductInterfaceQueryDefinition;
+import com.adobe.cq.commerce.magento.graphql.ProductPricesQueryDefinition;
+import com.adobe.cq.commerce.magento.graphql.ProductsQueryDefinition;
+import com.adobe.cq.commerce.magento.graphql.Query;
+import com.adobe.cq.commerce.magento.graphql.QueryQuery;
+import com.day.cq.wcm.api.Page;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import javax.annotation.PostConstruct;
+import javax.inject.Inject;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.models.annotations.Model;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Model(adaptables = SlingHttpServletRequest.class, adapters = ProductCarousel.class, resourceType = ProductCarouselImpl.RESOURCE_TYPE)
+public class ProductCarouselImpl implements ProductCarousel{
+    
+    protected static final String RESOURCE_TYPE = "venia/components/commerce/product/v1/productcarousel";
+    private static final Logger LOGGER = LoggerFactory.getLogger(ProductCarouselImpl.class);
+    
+    @Inject
+    private Resource resource;
+
+    @Inject
+    private String[] productSkuList;
+    
+    @Inject
+    private Page currentPage;
+    
+    private List<ProductInterface> productList;
+
+    private Page productPage;
+    
+    private MagentoGraphqlClient magentoGraphqlClient;
+    
+    @PostConstruct
+    private void initModel() {
+        final List<String> productKeys = Arrays.asList(this.productSkuList);
+        magentoGraphqlClient = MagentoGraphqlClient.create(resource);
+        productPage = Utils.getProductPage(currentPage);
+        if (productPage == null) {
+            productPage = currentPage;
+        }
+        
+        GraphqlClient client = resource.adaptTo(GraphqlClient.class);
```

Comment:
`GraphqlClient` is not needed anymore since you have `magentoGraphqlClient`. It is not used in the method you are calling anyway.

Reply:
code cleaned up.